src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/data_migration/BlogPostUidShortener.java

```diff
@@ -0,0 +1,34 @@
+package pl.tomaszdziurko.jvm_bloggers.blog_posts.data_migration;
+
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Component;
+
+import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPost;
+import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPostRepository;
+
+import java.util.List;
+
+@Component
+class BlogPostUidShortener {
+
+    private final BlogPostRepository blogPostRepository;
+
+    @Autowired
+    public BlogPostUidShortener(BlogPostRepository blogPostRepository) {
```

Comment:
This handcrafted constructor can be replaced with class level annotation [`@RequiredArgsConstructor(onConstructor = @__(@Autowired))`](https://projectlombok.org/api/lombok/RequiredArgsConstructor.html).